File: code/C#/src/OopSolidSample/Domain/GameBase.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Data.SQLite;

namespace OopSolidSample.Domain
{
    public abstract class GameBase
    {
        public static IList<Type> AvailableGameTypes = new[] {typeof (DrawFiveGame), typeof (BlackjackGame)};

        public abstract string Name { get; }
        public abstract DrawResult Draw();
        public abstract IList<HighScore> GetHighScores();
        public abstract void SaveScore(string name, int score);
    }
}

File: code/java/src/oopSolidSample/domain/GameBase.java
package oopSolidSample.domain;

import java.util.List;

public abstract class GameBase
{
    public static Class[] AvailableGameTypes = { DrawFiveGame.class, BlackjackGame.class };
    public abstract String getName();
    public abstract DrawResult Draw();
    public abstract List<HighScore> GetHighScores();
    public abstract void SaveScore(String name, int score);
}